### PR TITLE
Homework4: Natalia Mashliakevich

### PR DESCRIPTION
**Pull request by nataliasipenko**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

**Review by belovmd — CHANGES_REQUESTED** [review]
sanity checks failed - redundant files in pr

**Comment by nataliasipenko** [comment]
> sanity checks failed - redundant files in pr

amended

File: src/homework4/task2.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+Города
+
+Дан список стран и городов каждой страны. Затем даны названия городов.
+Для каждого города укажите, в какой стране он находится.
+"""
+
+
+# str1 = 'Russia Moscow Petersburg Novgorod Kaluga'
```
Comment (belovmd) [inline src/homework4/task2.py:9]: redundant comments

File: src/homework4/task2.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+dct = {}
+
+
+def make_a_dict(n):
+    while n > 0:
+        str_ = input('Введите название страны и городов:')
+        lst = list(str_.split(' '))
+        key, value = lst[0], lst[1:]
```
Comment (belovmd) [inline src/homework4/task2.py:23]: country, *cities = str_.split()

File: src/homework4/task2.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        dct[key] = value
```
Comment (belovmd) [inline src/homework4/task2.py:24]: not optimal data structure

File: src/homework4/task3.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+одновременно как в первом списке, так и во втором.
+"""
+
+
+# Поскольку в условии указано различных чисел (т.е. не одинаковых),
+# то выбираем уникальные элементы в обоих списках.
+lst1 = [1, 2, 3, 4, 5]
+lst2 = [4, 5, 6, 7, 8, 0]
+
+print(len(set(lst1) ^ set(lst2)))
```
Comment (belovmd) [inline src/homework4/task3.py:12]: task not solved it's intersection

File: src/homework4/task4.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+входит только в один из этих списков.
+"""
+
+
+# Поскольку в условии указано различных чисел (т.е. не одинаковых),
+# то выбираем уникальные элементы в одном из списков, например в первом.
+lst1 = [1, 2, 3, 4, 5]
+lst2 = [4, 5, 6, 7, 8, 0]
+
+print(len(set(lst1) - set(lst2)))
```
Comment (belovmd) [inline src/homework4/task4.py:12]: not solved it's symmetric difference

File: src/homework4/task3.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+"""
+Даны два списка чисел. Посчитайте, сколько различных чисел содержится
+одновременно как в первом списке, так и во втором.
+"""
+
+
+# Поскольку в условии указано различных чисел (т.е. не одинаковых),
```
Comment (belovmd) [inline src/homework4/task3.py:7]: redundant comments

File: src/homework4/task5.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
+"""
+Языки
+
+Каждый из N школьников некоторой школы знает Mi языков.
+Определите, какие языки знают все школьники и языки, которые знает
+хотя бы один из школьников.
+"""
+
+
+# Входные данные
```
Comment (belovmd) [inline src/homework4/task5.py:10]: inconsistent docstring - with """ at the beginning and # in the end. inaccurate.

File: src/homework4/task5.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    mi = str(input('Введите количество языков школьника: '))
+    str_ = input('Введите языки, который знает школьник через пробел:')
+    lst = list(str_.split(' '))
+    for lang in lst:
+        counts_dict[lang] = counts_dict.get(lang, 0) + 1
+    key, value = students, lst[:]
+    dct[key] = value
+    n -= 1
+
+# print(dct)
```
Comment (belovmd) [inline src/homework4/task5.py:49]: redundant comments

File: src/homework4/task6.py [hunk 9]
```diff
@@ -0,0 +1,11 @@
+"""
+Слова
+
+Во входной строке записан текст. Словом считается последовательность непробельных
+символов идущих подряд, слова разделены одним или большим числом пробелов или
+символами конца строки. Определите, сколько различных слов содержится в этом тексте.
+"""
+
+str_ = 'one, two     three four five.'
+print(set(str_.split()))
```
Comment (belovmd) [inline src/homework4/task6.py:10]: why do we need this print?

File: src/homework4/task7.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+"""
+a = 76
+b = 32
+
+while a and b:
+    if a > b:
+        a %= b
+    else:
+        b %= a
+print(max(a, b))
```
Comment (belovmd) [inline src/homework4/task7.py:14]: it's a cheat - you don't know where is result and use max

**Review by belovmd — APPROVED** [review]
6,5

**Pull request closed without merging**